sound/soc/sof/sof-of-dev.c

```diff
@@ -39,6 +39,37 @@ static void sof_of_probe_complete(struct device *dev)
 	pm_runtime_enable(dev);
 }
 
+int sof_of_parse(struct platform_device *pdev)
+{
+	struct snd_sof_pdata *sof_pdata = platform_get_drvdata(pdev);
+	struct device_node *np = pdev->dev.of_node;
+	struct snd_soc_of_mach *mach;
+	int ret;
+
+	mach = devm_kzalloc(&pdev->dev, sizeof(*mach), GFP_KERNEL);
+	if (!mach)
+		return -ENOMEM;
+
+	/* firmware-name is optional in DT */
+	of_property_read_string(np, "firmware-name", &sof_pdata->fw_filename);
+	ret = of_property_read_string(np, "tplg-name",
+				      &sof_pdata->tplg_filename);
+	if (ret < 0)
+		return ret;
+
+	ret = of_property_read_string(np, "machine-drv-name",
+				      &mach->drv_name);
```

Comment:
I thought the machine driver would be probed using DT? This feels like you would still be creating a platform device using the machine driver names, which we do in ACPI since we don't have a hook to probe the machine driver, but since you do in DT why is this necessary?

Reply:
@plbossart yes it will be probed using DT. We still need the machine driver name here because of this:

```
snd_sof_new_platform_drv
   -> snd_sof_new_platform_drv
```

The simplified normal code flow for DT is like this:
```
sof-of-dev.c:
-> sof_of_probe
     -> snd_sof_device_probe
         -> sof_probe_continue
            -> sof_machine_check
                 -> snd_sof_machine_select (EMPTY Operation on IMX8)
                          -> snd_sof_set_mach_params (EMPTY operation on IMX8
            -> snd_sof_new_platform_drv (intializes some fields of the platform driver)
            -> devm_snd_soc_register_component (registers the DSP platform driver).
           -> snd_sof_machine_register (this is an EMPTY operation on IMX8)
```

Reply:
> We still need the machine driver name here because of this:
> 
> ```
> snd_sof_new_platform_drv
>    -> snd_sof_new_platform_drv
>         pd->ignore_machine = drv_name;
> ```

Why don't we remove this? I can't recall what this .ignore_machine is supposed to do.

Reply:
@plbossart ignore_machine is very important for us. It is what tells that the FE DAI links in the machine driver should be ignored for SOF

Reply:
@ranj063 this is only needed for legacy machine drivers with statically defined FEs, why would it matter for a new OF/DT machine driver?

Reply:
@ranj063 @plbossart I spent a lot of time understanding what is the purpose of ignore_machine even sent some patches clarify the comments inside soc_check_tplg_fes.

I cannot tell you that I fully understand how to go on without this field. Besides ignore FE DAIs it also allows us to override BEs.

Actually, my entire machine driver descriptions starts with a regular link and soc_check_tplg_fes function converts it to a BE.

Also, this overrides link platforms name.

```  dai_link->platforms->name = component->name```

@lgirdwood maybe you can shed some light on the usage of ignore_machine.

Reply:
@dbaluta, @plbossart is right that for new machine drv, ignore_machine is not really important (even in the case of the new ACPI machine drivers for Intel platforms ex sof_rt5682.c). What we should do is may be set this in the snd_sof_machine_select op so that we restrict it to just the Intel legacy machine drivers

Reply:
@dbaluta if we avoid this ignore_machine thingy then we don't need all these definitions... I am lazy but less code is better in most cases :-)

Reply:
@plbossart agree with you on that. Will try to remove it and adapt simple-audio-card for this new situation.

Using ignore_machine I was able to create one normal link with simple-audio card and convert it to a BE inside soc_check_tplg_fes :).